netbox/ipam/forms/model_forms.py

```diff
@@ -240,6 +240,14 @@ class Meta:
             'tenant', 'description', 'comments', 'tags',
         ]
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # #18605: only filter VLAN select list if scope field is a Site
+        if scope_field := self.fields.get('scope', None):
+            if scope_field.queryset.model is not Site:
+                self.fields['vlan'].widget.attrs.pop('data-dynamic-params', None)
```

Comment:
This approach of tackling this in `PrefixForm.__init__()` seems to be the preferred approach, based on forms in the codebase.